mergin/cli.py

```diff
@@ -436,7 +436,7 @@ def pull(ctx):
                 new_transferred_size = job.transferred_size
                 bar.update(new_transferred_size - last_transferred_size)  # the update() needs increment only
                 last_transferred_size = new_transferred_size
-        pull_project_finalize(job)
+        pull_project_finalize(job, mc.username())
```

Comment:
Seeing this change now and realizing we have modified the API of pull_project_finalize() in the previous PR... may I ask you for a small change:
- we would store username into `PullJob` when it is created in `pull_project_async()`
- the `pull_project_finalize()` would return to having just one argument `job` - and the username would be extracted from there

Because there may be already some other apps using this API, and the previous changes would cause API break. Also, the general idea was that all context for a pull/push is encapsulated in the job object...